src/samples/plugins/ModularSessionKeyPlugin.sol

```diff
@@ -0,0 +1,336 @@
+        bytes4[] calldata allowedSelectors,
+        uint48[] calldata validAfters,
+        uint48[] calldata validUntils
+    ) external {
+        if (
+            tempOwners.length != allowedSelectors.length || tempOwners.length != validAfters.length
+                || tempOwners.length != validUntils.length
+        ) {
+            revert WrongDataLength();
+        }
+        for (uint256 i = 0; i < tempOwners.length; i++) {
+            _addSessionKey(msg.sender, tempOwners[i], allowedSelectors[i], validAfters[i], validUntils[i]);
+        }
+        emit SessionKeysAdded(msg.sender, tempOwners, allowedSelectors, validAfters, validUntils);
+    }
+
+    function removeSessionKeyBatch(address[] calldata tempOwners, bytes4[] calldata allowedSelectors) external {
+        if (tempOwners.length != allowedSelectors.length) {
+            revert WrongDataLength();
+        }
+        for (uint256 i = 0; i < tempOwners.length; i++) {
+            _removeSessionKey(msg.sender, tempOwners[i], allowedSelectors[i]);
+        }
+        emit SessionKeysRemoved(msg.sender, tempOwners, allowedSelectors);
+    }
+
+    // ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
+    // ┃    Plugin view functions    ┃
+    // ┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
+
+    /// @inheritdoc IModularSessionKeyPlugin
+    function getSessionDuration(address account, address tempOwner, bytes4 allowedSelector)
+        external
+        view
+        returns (uint48 _validAfter, uint48 _validUntil)
+    {
+        bytes memory key = account.allocateAssociatedStorageKey(0, 1);
+        StoragePointer ptr = key.associatedStorageLookup(keccak256(abi.encodePacked(tempOwner, allowedSelector)));
+        SessionInfo storage sessionInfo = _castPtrToStruct(ptr);
+        _validAfter = sessionInfo.validAfter;
+        _validUntil = sessionInfo.validUntil;
+    }
+
+    // ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
+    // ┃    Plugin interface functions    ┃
+    // ┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
+
+    /// @inheritdoc BasePlugin
+    function onInstall(bytes calldata data) external override {
+        if (data.length != 0) {
+            (
+                address[] memory tempOwners,
+                bytes4[] memory allowedSelectors,
+                uint48[] memory validAfters,
+                uint48[] memory validUntils
+            ) = abi.decode(data, (address[], bytes4[], uint48[], uint48[]));
+            if (
+                tempOwners.length != allowedSelectors.length || tempOwners.length != validAfters.length
+                    || tempOwners.length != validUntils.length
+            ) {
+                revert WrongDataLength();
+            }
+            for (uint256 i = 0; i < tempOwners.length; i++) {
+                _addSessionKey(msg.sender, tempOwners[i], allowedSelectors[i], validAfters[i], validUntils[i]);
+            }
+        }
+    }
+
+    /// @inheritdoc BasePlugin
+    function onUninstall(bytes calldata data) external override {
```

Comment:
Though it will add some gas overhead, this sounds necessary to keep the consistency in the storage of plugin. I added an additional mapping that stores the information of session key and allowed selector to implement this.